src/contrib/cache_req.rst

```diff
@@ -0,0 +1,290 @@
+.. contents:: Table of Contents
+    :local:
+
+Cache Req
+=========
+
+The ``cache req`` (request) interface provides an internal API to request identity
+information from SSSD, either directly from the local SSSD cache or retrieving refreshed data
+from the data provider. The SSSD NSS responder is the primary consumer of the interface,
+however ``cache_req`` also refreshes information which may be needed by the PAM
+responder, or provide user information attributes over the SSSD infopipe.
+
+.. mermaid::
+
+	graph LR
```

Comment:
This chart looks "too flat" to me. I think similar to the one [1] made by jhrozek in the past would be more meaningful.
[1] https://jhrozek.files.wordpress.com/2015/03/sssd-lookup.png